OsmAnd/src/net/osmand/plus/poi/PoiHelper.java

```diff
@@ -0,0 +1,321 @@
+package net.osmand.plus.poi;
+
+
+import android.annotation.SuppressLint;
+import android.os.AsyncTask;
+
+import androidx.annotation.NonNull;
+
+import net.osmand.PlatformUtil;
+import net.osmand.binary.BinaryMapIndexReader;
+import net.osmand.osm.MapPoiTypes;
+import net.osmand.osm.PoiCategory;
+import net.osmand.osm.PoiType;
+import net.osmand.plus.OsmandApplication;
+import net.osmand.plus.activities.LocalIndexHelper;
+import net.osmand.plus.activities.LocalIndexInfo;
+import net.osmand.plus.api.SQLiteAPI;
+import net.osmand.plus.download.ui.AbstractLoadLocalIndexTask;
+
+import org.apache.commons.logging.Log;
+import org.json.JSONArray;
+import org.json.JSONException;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class PoiHelper {
+
+	private static final Log LOG = PlatformUtil.getLog(PoiHelper.class);
+	private OsmandApplication app;
+	private PoiDbHelper helper;
+	private Map<String, Long> files;
+	private Map<String, List<String>> categories;
+
+	public PoiHelper(OsmandApplication app) {
+		this.app = app;
+		helper = new PoiDbHelper(app);
+	}
+
+	public Map<String, Long> getFiles() {
+		if (files == null) {
+			files = helper.getFiles();
+		}
+		return files;
+	}
+
+	public Map<String, List<String>> getCategories() {
+		if (categories == null) {
+			categories = helper.getCategories();
+		}
+		return categories;
+	}
+
+	public void readPoiTypesFromMap() {
+		LocalIndexHelper localIndexHelper = new LocalIndexHelper(app);
+		List<LocalIndexInfo> localMapsIndexes = localIndexHelper.getLocalFullMaps(new AbstractLoadLocalIndexTask() {
```

Comment:
Bad link, Helpers/Serice connected to OsmAndApplication shouldn't use LocalIndexInfo/LocalIndexHelper which are UI classes at all. Static methods are also bad